content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx

```diff
@@ -0,0 +1,186 @@
+
+## Constraint and index names (Preview)
+
+In [2.29.0](https://github.com/prisma/prisma/releases/tag/2.29.0) and later, you can use the `map` field to define the **underlying constraint and index names** in the Prisma schema. The `map` field is _only_ visible in the schema if you use a constraint name that differs from Prisma's naming convention.
+
+<Admonition type="warning">
+
+**Not supported by MongoDB**<br />
+Custom constraint and index names apply to relational databases only.
+
+</Admonition>
+
+### Enabling the <inlinecode>namedConstraints</inlinecode> Preview feature
+
+Named constraints is a Preview feature. To enable this feature:
+
+1. Add `namedConstraints` to the `previewFeatures` field in the `generator` block:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["namedConstraints"]
+}
+```
+
+2. Choose an [upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2290#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
+
+<Admonition type="alert">
+
+If you use Prisma Migrate and want to maintain your existing constraint and index names, **do not** enable the `namedConstraints` feature and immediately run `prisma migrate dev`. This will **change any underlying constraint name that does not follow Prisma's convention**. Follow the [upgrade path for the Prisma Migrate workflow](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2290#upgrade-paths-for-prisma-migrate-users).
+
+</Admonition>
+
+### Use cases for named constraints
+
+Some use cases for explicitly named constraints include:
+
+- Company policy
+- Conventions of other tools
+
+### Using custom constraint names
+
+The following schema defines three constraints (`@id`, `@unique`, and `@relation`) and one index (`@@index`):
+
+```prisma highlight=2,8,11,14;normal
+model User {
+  id    Int    @id @default(autoincrement())
+  name  String @unique
+  posts Post[]
+}
+
+model Post {
+  id          Int     @id @default(autoincrement())
+  title       String
+  authorName  String  @default("Anonymous")
+  author      User?   @relation(fields: [authorName], references: [name])
+}
+
+@@index([title, authorName])
+```
+
+By default, constraint and index names in the underlying database follow Prisma's naming convention. Constraint or index names that follow Prisma's naming convention are not included in the schema.
+
+<Admonition type="info">
+
+See the [attributes reference documentation](../../../reference/api-reference/prisma-schema-reference#attributes) <span class="api"></span> for default constraint and index naming conventions - for example: [`@unique`](../../../reference/api-reference/prisma-schema-reference#arguments-3) <span class="api"></span>
```

Comment:
Hm this is a bit weird as it links to a whole headline and finding that information in there will be super hard. I could not, although I know what I am looking for. (Should there be headlines "Naming" there? Not sure)

Reply:
I also did not see our whole default naming convention spelled out. That's another todo I added.

Reply:
removed this